commit 5b906ce6eb02118697c2f81d462ddfa724377fe8
Author: Will Fitch <willfitch@php.net>
Date:   Sat Jan 18 11:25:53 2014 -0500

    Fix bug #66509: copy() arginfo incorrect since 5.4
    
    Since 5.4, the ZEND_BEGIN_ARG_INFO_EX was replaced
    by non _EX, causing Reflection to assume the
    $context parameter is required.

--- php5.orig/ext/standard/basic_functions.c
+++ php5/ext/standard/basic_functions.c
@@ -1206,7 +1206,7 @@ ZEND_END_ARG_INFO()
 ZEND_BEGIN_ARG_INFO(arginfo_fstat, 0)
 	ZEND_ARG_INFO(0, fp)
 ZEND_END_ARG_INFO()
-ZEND_BEGIN_ARG_INFO(arginfo_copy, 0)
+ZEND_BEGIN_ARG_INFO_EX(arginfo_copy, 0, 0, 2)
 	ZEND_ARG_INFO(0, source_file)
 	ZEND_ARG_INFO(0, destination_file)
 	ZEND_ARG_INFO(0, context)
--- /dev/null
+++ php5/ext/standard/tests/file/bug66509.phpt
@@ -0,0 +1,15 @@
+--TEST--
+Bug #66509 (copy() showing $context parameter as required)
+--FILE--
+<?php
+
+$r = new \ReflectionFunction('copy');
+ 
+foreach($r->getParameters() as $p) {
+    var_dump($p->isOptional());	
+}
+?>
+--EXPECT--
+bool(false)
+bool(false)
+bool(true)
